src/promptflow/promptflow/_sdk/_serving/utils.py

```diff
@@ -137,3 +142,48 @@ def encode_dict(data: dict) -> str:
     b64_data = base64.b64encode(zipped_data)
     # bytes -> str
     return b64_data.decode()
+
+
+def try_extract_trace_context(logger) -> Context:
+    """Try to extract trace context from request headers."""
+    # reference: https://www.w3.org/TR/trace-context/
+    trace_parent = request.headers.get('Traceparent')
+    if trace_parent:
+        carrier = {'traceparent': trace_parent}
+        trace_state = request.headers.get('Tracestate')
+        if trace_state:
+            carrier['tracestate'] = trace_state
+        ctx = TraceContextTextMapPropagator().extract(carrier=carrier)
+    else:
+        ctx = None
+    baggage = request.headers.get('Baggage')
+    if baggage:
+        b2 = {'baggage': baggage}
+        baggage_ctx = W3CBaggagePropagator().extract(b2, context=ctx)
+    else:
+        baggage_ctx = ctx
+    if baggage_ctx:
+        logger.info(f"Received trace context: {baggage_ctx}")
+    return baggage_ctx
+
+
+def serialize_attribute_value(v):
```

Comment:
Do we already have a similar method in the `promptflow.tracing` package? cc @lumoslnt @thy09

Reply:
this is user's data, we need to try our best to keep it as is.